### DISCUSSION TITLE
JS Field date render from ISO

### DISCUSSION THREAD

**Discussion by au8st** [Budibase/budibase#13008] [Help] [answered]
Hi all,

Learning and very new to js, from examples from the budibase documentation and IF ELSEIF statements I can't figure out how to `truncate` or `reformat` a date from `2023-07-08T00:00:00.000Z` as example.

The column js formula looks for a reminder column selection and when yes this takes the data from the type of service and adds a certain amount of days to the reminder:

```js
if ($("Service Reminder")=="No"){
  return "N/A";
} else if ($("Service Type")=="SERVICE1"){
    const date = new Date($("Date Of Service"));
date.setDate(date.getDate() + 14);
return date;
} else if ($("Service Type")=="SERVICE2"){
    const date = new Date($("Date Of Service"));
date.setDate(date.getDate() + 182);
return date;
} else if ($("Service Type")=="SERVICE3"){
    const date = new Date($("Date Of Service"));
date.setDate(date.getDate() + 365);
return date;
} else if ($("Service Type")=="SERVICE4"){
    const date = new Date($("Date Of Service"));
date.setDate(date.getDate() + 28);
return date;
}
```

When also pulling the date in handlebars the date is also formatted in ISO e.g `2023-07-08T00:00:00.000Z`, how could I display this as `2023-07-08`?

Referencing [Issue 10730](https://github.com/Budibase/budibase/discussions/10730) when I set the `const dateString = $("Date Of Service");` this will simply replicate the date in the selected field and ignore the date addition.

Thanks in advance

**Comment by ConorWebb96** [comment, marked as answer]
Hey @au8st,

Let me know if the below works for you, if it doesn't I'll need an **app export** to replicate your exact settings for testing.

```js
function formatDate(date) {
  const year = date.getFullYear().toString().slice(-2);
  const month = (date.getMonth() + 1).toString().padStart(2, '0');
  const day = date.getDate().toString().padStart(2, '0');
  return `${year}-${month}-${day}`;
}

function calculateDueDate(serviceType, dateOfService) {
  const date = new Date(dateOfService);

  switch (serviceType) {
    case "SERVICE1":
      date.setDate(date.getDate() + 14);
      break;
    case "SERVICE2":
      date.setDate(date.getDate() + 182);
      break;
    case "SERVICE3":
      date.setDate(date.getDate() + 365);
      break;
    case "SERVICE4":
      date.setDate(date.getDate() + 28);
      break;
    default:
      return "N/A";
  }

  return formatDate(date);
}

const serviceType = $("Service Type");
const dateOfService = $("Date Of Service");

if ($("Service Reminder") == "No") {
  return "N/A";
} else {
  const dueDate = calculateDueDate(serviceType, dateOfService);
  return dueDate;
}
```

Hopefully this helps.

**Reply by au8st** [reply]
Connor that is brilliant thank you kindly, it's also a good insight to learn javascript further as an insight to see how things actually work and learn from my new understanding and mistakes.

**Reply by ConorWebb96** [reply]
Just for your information the reason I used switches here instead of if else, is because switches scale better. Say if you have a **service5** and it has the same increment to the date as service1 you can just add an additional case below it without having to write additional code.

No problem, I'm glad this helped you. 🚀

**Reply by au8st** [reply]
Another new thing learned, thank you again, it's greatly appreciated :)
